src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
 
     @Override
     public void put(K key, V value) {
-
+        checkCapacity();
+        Node<K, V> node = new Node<>(key, value);
+        int index = getHash(key);
+        if (table[index] == null) {
+            table[index] = node;
+        } else {
+            Node<K, V> previousNode = table[index];
+            while (previousNode != null) {
+                if (Objects.equals(key, previousNode.key)) {
+                    previousNode.value = value;
+                    return;
+                }
+                if (previousNode.next == null) {
+                    previousNode.next = node;
+                    break;
+                }
+                previousNode = previousNode.next;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        for (int i = 0; i < capacity; i++) {
```

Comment:
I think this is too complicated.
Look, you have method for finding bucket index(getHash).
You could just do smth like this:

Node<K, V> current = table[getHash(key)];
        while (current != null) {
            if (Objects.equals(current.key, key)) {
                return current.value;
            }
            current = current.next;
        }
        return null;

Just use you method getHash ;)